src/Agent.Listener/Configuration/ServiceControlManager.cs

```diff
@@ -60,6 +60,11 @@ protected void CalculateServiceName(AgentSettings settings, string serviceNamePa
         protected void SaveServiceSettings()
         {
             string serviceConfigPath = IOUtil.GetServiceConfigFilePath();
+            if(File.Exists(serviceConfigPath))
+            {
+                IOUtil.DeleteFile(serviceConfigPath);
```

Comment:
Looks like since we set the file attribute to hiden, when overwrite i got permission exception.